core/src/main/scala/scalafix/util/ChangeMethod.scala

```diff
@@ -0,0 +1,41 @@
+package scalafix.util
+
+import scala.collection.immutable.Seq
+import scala.{meta => m}
+import scalafix.rewrite.{Desugared, RewriteCtx, SemanticApi}
+
+case class ReplaceTerm(original: m.Type, newString: String)
+
+class ChangeMethod(ast: m.Tree, ctx: RewriteCtx)(implicit sApi: SemanticApi) {
+  object DType extends Desugared[m.Type]
+  object DTerm extends Desugared[m.Term]
+
+  //This isn't very satisfying.  I am trying to match on a method, without including
+  //it's arguments
+  def partialTermMatch(rt: ReplaceTerm)
+    : PartialFunction[m.Tree, (scala.meta.Term, ReplaceTerm)] = {
+    case t @ DTerm(desugared)
+        if desugared.syntax.startsWith(rt.original.syntax) &&
+          desugared.syntax.endsWith("]") =>
```

Comment:
Are you doing this to overcome the inserted `.apply[T]` in the desugared output? I matched on the `.apply` in the rewrite: https://github.com/olafurpg/scalafix/blob/445b2cd783ef9d1e041198663ed335747133dfcc/core/src/main/scala/scalafix/rewrite/Xor2Either.scala#L22
Maybe I was too optimistic about the capabilities of this `desugared` approach. I am afraid this `desugared` approach will always require awkward workarounds like this.

In the future, I'm aiming for a user experience where the user can configure in `.scalafix.conf`

```
rename = [
  { from = "cats.data.Xor", to = "scala.util.Either }
]
```
and everything just works™️. But I suspect we'll need to get type equality `=:=`  in the semantic api to do that.